xarray/core/coordinates.py

```diff
@@ -241,13 +456,115 @@ def merge(self, other: Coordinates | None) -> Dataset:
             variables=coords, coord_names=coord_names, indexes=indexes
         )
 
+    def merge_coords(self, other: Mapping[Any, Any] | None = None) -> Coordinates:
```

Comment:
I'm not sure this name is worth the trouble, despite the better API. `coords.merge(other).coords` is at least as succinct as `coords.merge_coords(other)`.

Instead, we could consider a deprecation cycle to remove the existing `.merge()`, or perhaps even change the return type from `Dataset` to `Coordinates`? That would be a breaking change, but also one with very low impact. There are very few uses of `Coordinates.merge` currently and all of those could be replaced by calling `xarray.merge()`.

Reply:
Agreed.

I added `merge_coords` before adding support for `Coordinates` objects in `xarray.merge()` and `xarray.align()`. It's better to reuse those instead of a new method.

> or perhaps even change the return type from Dataset to Coordinates?

Yes that would make sense.

This makes me also think about what to do when a Dataset object is passed as `coords` argument? I think that currently it is handled like a regular mapping (indexes are ignored). Is it what we want or should we treat it like `Coordinates` and extract indexes too?